storage/storage-interface/src/state_view.rs

```diff
@@ -182,18 +165,59 @@ impl StateView for VerifiedStateView {
             }
         };
 
+        Ok(state_store_value_option)
+    }
+}
+
+pub struct StateCache {
+    pub frozen_base: FrozenSparseMerkleTree<StateValue>,
+    pub accounts: HashMap<AccountAddress, AccountState>,
+    pub proofs: HashMap<HashValue, SparseMerkleProof<StateValue>>,
+}
+
+impl StateView for VerifiedStateView {
+    fn id(&self) -> StateViewId {
+        self.id
+    }
+
+    fn get_state_value(&self, state_key: &StateKey) -> Result<Option<Vec<u8>>> {
+        // First check if the cache has the state value.
+        if let Some(contents) = self.state_key_value_cache.read().get(state_key) {
+            return Ok(Some(contents.bytes.clone()));
+        }
+        let state_store_value_option = self.get_state_value_internal(state_key)?;
+        // Update the cache if still empty
+        let mut cache = self.state_key_value_cache.write();
+        let new_value = cache
+            .entry(state_key.clone())
+            .or_insert_with(|| state_store_value_option.unwrap_or_default());
+
+        Ok(Some(new_value.bytes.clone()))
+    }
+
+    fn get_account_resource(&self, access_path: &AccessPath) -> Result<Option<Vec<u8>>> {
```

Comment:
This works for modules as well, so the name isn't correct?

Reply:
Good point, any suggestion on name?

Reply:
`get_by_access_path()` :)